docs/spec_draft.md

```diff
@@ -1420,6 +1421,79 @@ Produces an `output` tensor from a constant `value`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.convert
+
+### Semantics
+
+Performs an element-wise conversion of values similar to an element-wise
+`static_cast` in C++ from one element type to another on `operand` tensor and
+produces a `result` tensor.
+
+For conversions involving **integer-to-integer**, if there is an unsigned/signed
+overflow, the result is implementation-defined and one of the following:
+  * mathematical result modulo $2^n$, where n is the bit width of the result,
+    for unsigned overflow. For signed integer overflow, wraps the result around
+    the representable range $[-2^{n-1},\ 2^{n-1} - 1]$.
+  * saturation to $2^{n-1} - 1$ (or $-2^{n-1}$) for signed overflow and
+    saturation to $2^n - 1$ (or $0$) for unsigned overflow.
+
+For conversions involving **floating-point-to-floating-point** or
+**integer-to-floating-point**, if the source value can be exactly represented in
+the destination type, the result value is that exact representation. If the
+source value is between two adjacent destination values, the result is
+implementation-defined. Otherwise, the behavior is undefined.
+
+Conversion involving **complex-to-complex** follows the same behavior of
+**floating-point-to-floating-point** conversions for converting the inner
+representation.
+
+For conversions involving **floating-point-to-complex** or
+**complex-to-floating-point**, the destination imaginary value is zeroed or the
+source imaginary value is ignored, respectively. The conversion of inner
+representation of complex element type follows the
+**floating-point-to-floating-point** conversion.
+
+Conversions involving **integer-to-complex** follows the same behavior as
+**integer-to-floating-point** conversion while converting the source integer to
+destination real part. The destination imaginary part is zeroed.
+
+For conversions involving **floating-point-to-integer**, the fractional part is
+truncated. The behavior is undefined if the truncated value cannot be
+represented in the destination type. Conversions involving
+**complex-to-integer** follows the same behavior while converting the source
+real part to destination integer. The source imaginary part is ignored.
+
+For **boolean-to-any-supported-type** conversions, the value `false` is
+converted to zero, and the value `true` is converted to one. For
+**any-supported-type-to-boolean** conversions, a zero value is converted to
+`false` and any non-zero value is converted to `true`.
```

Comment:
Unfortunately, I don't have the time right now to run comprehensive experiments on my own, so I wanted to check in. When writing the spec, have you got a chance to check each supported configuration of operand/result types via `run_hlo_module`?

Reply:
I've done most of the testing for different configurations but definitely not exhaustive (17^2 combinations!), and specifically for booleans, I've confirmed that boolean to any type and any type to boolean behaves as noted in the spec. I think we should also leave this to #223.

Reply:
Agreed, let's leave this to #223.